tests/autoyast/prepare_profile.pm

```diff
@@ -20,79 +20,33 @@ use strict;
 use warnings;
 use base "opensusebasetest";
 use testapi;
-use version_utils 'is_sle';
-use registration 'scc_version';
-use autoyast 'expand_template';
-use File::Copy 'copy';
-use File::Path 'make_path';
+use autoyast qw(
+  detect_profile_directory
+  expand_template
+  expand_version
+  adjust_network_conf
+  expand_variables
+  upload_profile);
 
 sub run {
     my $path = get_required_var('AUTOYAST');
     # Get file from data directory
     my $profile = get_test_data($path);
 
-    # try to detect directory (autoyast_opensuse/, autoyast_sle{12,15}/, autoyast_sles11/)
-    # TODO: autoyast_{caasp,kvm,qam,xen}
-    my $dir    = "autoyast_";
-    my $regexp = $dir . '\E[^/]+\/';
+    $path    = detect_profile_directory(profile => $profile, path => $path);
+    $profile = get_test_data($path);
+    die "Empty profile" unless $profile;
```

Comment:
just to make sure. if the $path is not found this would fail in some other point? Do you know? because detect_profile_directory doesnt die anywhere. So the die here can be anything than "Empty profile".

Reply:
I did the refactor just to avoid to introduce new setting, totally out of scope for me to enhance error handling of the library :) I just added "Empty profile" instead of previous code that just died.